web/src/app/donate/DonateComponent.js

```diff
@@ -0,0 +1,193 @@
+import {
+  Button,
+  FormControl,
+  FormGroup,
+  ControlLabel,
+  InputGroup,
+  Grid,
+  Row,
+  Col
+} from 'react-bootstrap';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import PayWithPayPalComponent from './PayWithPayPalComponent';
+import styles from './DonateStyles.scss';
+
+class DonateComponent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: '',
+      amount: '',
+      bounty: '',
+      nameError: null,
+      amountError: null,
+      bountyError: null
+    };
+
+    this.submitForm = this.submitForm.bind(this);
+    this.validateAmount = this.validateAmount.bind(this);
+  }
+
+  submitForm(e) {
+    e.preventDefault();
+    this.validateName();
+    this.validateAmount();
+    this.validateBounty();
+
+    if ((this.state.nameError && this.state.nameError.type === null) &&
+        (this.state.amountError && this.state.amountError.type === null) &&
+        (this.state.bountyError && this.state.bountyError.type === null)) {
+          // everything is valid
+          const payload = {
+            username: this.state.username,
+            amount: this.state.amount,
+            bounty: this.state.bounty
+          };
+          this.props.insertBounty(payload);
+    }
+  }
+
+  validateName() {
+    let error = { type: null };
+    if (this.state.username === '') {
+      error = { type: 'error', 'message' : 'This field is required' };
+    }
+
+    this.setState({
+      nameError: error
+    });
+  }
+
+  validateAmount() {
```

Comment:
we'll need to check for length (no more than 2 decimals)
we'll need to check for max and min values (no value less than $1, no value greater than $1000.... these are arbitrary and can get finalized values later)